WooCommerce/Classes/ViewModels/CardPresentPayments/CardPresentModalConnectionFailedUpdateAddress.swift

```diff
@@ -26,13 +36,52 @@ final class CardPresentModalConnectingFailedUpdateAddress: CardPresentPaymentsMo
 
     let bottomSubtitle: String? = nil
 
-    init(continueSearch: @escaping () -> Void, cancelSearch: @escaping () -> Void) {
-        self.continueSearchAction = continueSearch
+    init(adminUrl: URL?, site: Site?, retrySearch: @escaping () -> Void, cancelSearch: @escaping () -> Void) {
+        self.adminUrl = adminUrl
+        self.site = site
+        self.retrySearchAction = retrySearch
         self.cancelSearchAction = cancelSearch
     }
 
     func didTapPrimaryButton(in viewController: UIViewController?) {
-        continueSearchAction()
+        guard let adminUrl = adminUrl,
+              let viewController = viewController else {
+            return retrySearchAction()
+        }
+        switch site?.isWordPressStore {
+        case true:
+            presentAuthenticatedWebview(url: adminUrl, from: viewController)
+        default:
+            UIApplication.shared.open(adminUrl)
+            // TODO: Replace button with refresh? Dismiss the alert? Do the retry when the user returns to the app?
```

Comment:
I would suggest keeping it as simple as possible in the first iteration. So replacing the button with "refresh" sounds good enough for now?